packit/copr_helper.py

```diff
@@ -199,11 +181,21 @@ def get_fields_to_change(
     ) -> Dict[str, Tuple[Any, Any]]:
 
         fields_to_change: Dict[str, Tuple[Any, Any]] = {}
-        if chroots is not None and set(chroots) != set(copr_proj.chroot_repos.keys()):
-            fields_to_change["chroots"] = (
-                list(copr_proj.chroot_repos.keys()),
-                chroots,
-            )
+        if chroots is not None:
+            old_chroots = set(copr_proj.chroot_repos.keys())
+
+            new_chroots = None
+            if copr_proj.ownername == "packit" and set(chroots) != old_chroots:
+                new_chroots = chroots
+            elif not set(chroots).issubset(old_chroots):
+                new_chroots = list(set(chroots) | old_chroots)
+
+            if new_chroots:
+                new_chroots.sort()
+                fields_to_change["chroots"] = (
+                    list(old_chroots),
+                    new_chroots,
+                )
```

Comment:
I'm wondering if we actually need to do this `if ownername == packit`, since we can just request a build for the specific chroot set and don't touch the project settings

https://python-copr.readthedocs.io/en/latest/client_v3/build_options.html

Reply:
Lemme have a look, but I think we set them either way. This just happens before running the build, we fetch the Copr repo and (if necessary) update the chroots…

Reply:
and that's my point :D we should not update the chroot setting of the project, we should request a build for specific chroots only if `owner != packit`

Reply:
That would duplicate the handling of Copr exceptions, cause currently if you want to build for unset chroots, we try to set them and in case we fail, report them. This would fail when running the build, I guess it would end up in Sentry.

Reply:
One way or another it's still a breaking change if someone **currently** expects Packit to „purge“ the unused chroots on their own Copr repos.